packages/components/root/stories/root.stories.js

```diff
@@ -1,125 +1,135 @@
-		() => {
-			const root = createRoot();
-
-			createElement(slots.header, root);
-			createElement(slots.footer, root);
-			createElement(slots['sidebar-left'], root);
-			createElement(slots['sidebar-right'], root);
-			createElement(slots['sidebar-inner-left'], root);
-			createElement(slots['sidebar-inner-right'], root);
-
-			return root;
+		if (header) {
+			enableSlot('header');
+		} else {
+			disableSlot('header');
+		}
+		if (footer) {
+			enableSlot('footer');
+		} else {
+			disableSlot('footer');
 		}
-	);
+		if (sidebarLeft) {
+			enableSlot('sidebar-left');
+		} else {
+			disableSlot('sidebar-left');
+		}
+		if (sidebarRight) {
+			enableSlot('sidebar-right');
+		} else {
+			disableSlot('sidebar-right');
+		}
+		if (sidebarInnerLeft) {
+			enableSlot('sidebar-inner-left');
+		} else {
+			disableSlot('sidebar-inner-left');
+		}
+		if (sidebarInnerRight) {
+			enableSlot('sidebar-inner-right');
+		} else {
+			disableSlot('sidebar-inner-right');
+		}
+		return root;
+	});
 
 function createRoot() {
-	const root = document.createElement('ts-root');
-	root.innerHTML = 'Content';
-	root.setAttribute('style', 'display: block; width: 100%; height: 100%;');
-	return root;
+	return html`
+		<ts-root style="display: block; width: 100%; height: 100%;">
+			<header
+				slot="!header"
+				style="height:60px;background:#00AEFF"
+				class="header"
+			>
+				Header
+			</header>
+			<footer
+				slot="!footer"
+				style="height:60px;background:#9AB2BC"
+				class="footer"
+			>
+				Footer
+			</footer>
+			<section
+				slot="!sidebar-left"
+				style="height:auto;width:320px;background:#50C610"
+				class="sidebar-left"
+			>
+				Left Sidebar
+			</section>
+			<section
+				slot="!sidebar-right"
+				style="height:auto;width:320px;background:#FDBE12"
+				class="sidebar-right"
+			>
+				Right Sidebar
+			</section>
+			<section
+				slot="!sidebar-inner-left"
+				style="height:auto;width:320px;background:#A70262"
+				class="sidebar-inner-left"
+			>
+				Inner Left Sidebar
+			</section>
+			<section
+				slot="!sidebar-inner-right"
+				style="height:auto;width:320px;background:#90129B"
+				class="sidebar-inner-right"
+			>
+				Inner Right Sidebar
+			</section>
+		</ts-root>
+	`;
 }
 
-function createElement({ slot, height, background, width }, root) {
-	const tag = ['header', 'footer'].includes(slot) ? slot : 'div';
-	const element = document.createElement(tag);
-	element.innerHTML = slot;
-	element.setAttribute('slot', slot);
-	element.setAttribute('class', slot);
-	element.setAttribute(
-		'style',
-		`height: ${height}; background: ${background}; ${
-			width ? `width: ${width};` : ''
-		}`
-	);
-	root.appendChild(element);
+function enableSlot(name) {
+	var disabledSlot = document.querySelector('[slot="!' + name + '"]');
+	if (disabledSlot) {
+		return disabledSlot.setAttribute('slot', name);
+	}
+}
+
+function disableSlot(name) {
+	var enabledSlot = document.querySelector('[slot=' + name + ']');
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "var" with "let" or "const" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS3504)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert this concatenation to the use of a template. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS3512)